### PR TITLE
feat: add '--merge' option when deleting branch

### LINKED ISSUE #2917: Provide option for deleting merged branches
### Summary 

Providing an option to delete merged branches from a repo. They are not needed anymore and just distract.

### Steps to reproduce the behavior

Using JenkinsX a lot of branches get created for pull requests which could be deleted. For example https://github.com/jenkins-x/cloud-environments has currently 58 branches. Arguably the deletion of a merged branch can be automated, but I think it would be nice to have this as a dedicated option as well, especially since there is already a `jx delete branch`.

### Expected behavior

Provide something like `jx delete branch -o my-org -n my-repo --merged` which deletes all merged branches.

### Actual behavior

ATM there is no such option.

### Jx version

The output of `jx version` is:

```
jx                 1.3.807
jenkins x platform 0.0.3320
Kubernetes cluster v1.10.11-gke.1
kubectl            v1.10.7
helm client        v2.10.0+g9ad53aa
helm server        v2.12.2+g7d2b0c7
git                git version 2.17.2 (Apple Git-113)
Operating System   Mac OS X 10.14.2 build 18C54
```

### Jenkins type

- [x] Classic Jenkins
- [x] Serverless Jenkins

### Kubernetes cluster

GKE

### Operating system / Environment

macOS 10.14

### PR DESCRIPTION
**Pull request by mrageh**
#### Submitter checklist

- [x] Change is code complete and matches issue description.
- [ ] Change is covered by existing or new tests.

#### Description
Adds new `--merge` option to the `jx delete branch` command. Which will
delete all merged branches on a repository.

#### Which issue this PR fixes

fixes #2917

### CODE REVIEW

**Comment by hferentschik** [comment]
@mrageh, thanks for the pull request. You will need to run `make generate-mocks` to make sure the generated mock interfaces are up to date. If you run this make target you should see that several files gets updated. Add these file to this commit and that should take care of the `lint` status check failing.

File: pkg/cmd/deletecmd/delete_branch.go [hunk 1]
```diff
@@ -79,12 +80,12 @@ func NewCmdDeleteBranch(commonOpts *opts.CommonOptions) *cobra.Command {
 	cmd.Flags().StringVarP(&options.SelectFilter, "filter", "f", "", "If selecting branches to remove this filters the list of repositories")
 	cmd.Flags().BoolVarP(&options.SelectAllRepos, "all-repos", "", false, "If selecting projects to remove branches this defaults to selecting them all")
 	cmd.Flags().StringVarP(&options.SelectFilterRepos, "filter-repos", "", "", "If selecting projects to remove brancehs this filters the list of repositories")
+	cmd.Flags().BoolVarP(&options.Merged, "merged", "", false, "If deleting merged branches in a repository")
```
Comment (hferentschik) [inline pkg/cmd/deletecmd/delete_branch.go]: 👍

File: pkg/cmd/deletecmd/delete_branch.go [hunk 2]
```diff
@@ -154,48 +155,89 @@ func (o *DeleteBranchOptions) Run() error {
...
-		if !o.BatchMode {
-			log.Logger().Warnf("You are about to delete these branches '%s' on the Git provider. This operation CANNOT be undone!",
-				strings.Join(branches, ","))
-
-			flag := false
-			prompt := &survey.Confirm{
-				Message: "Are you sure you want to delete these all these branches?",
-				Default: false,
+		if o.Merged {
+			branches, err := o.Git().RemoteMergedBranchNames(dir, "remotes/origin/")
```
Comment (hferentschik) [inline pkg/cmd/deletecmd/delete_branch.go]: :-)

File: pkg/gits/git_cli.go [hunk 3]
```diff
@@ -784,8 +784,23 @@ func (g *GitCLI) GetRemoteUrl(config *gitcfg.Config, name string) string {
 
 // RemoteBranches returns all remote branches with the given prefix
 func (g *GitCLI) RemoteBranchNames(dir string, prefix string) ([]string, error) {
+	return g.remoteBranchNames(dir, prefix, false)
+}
+
+// RemoteMergedBranchesNames returns all remote branches that were merged with the given prefix
+func (g *GitCLI) RemoteMergedBranchNames(dir string, prefix string) ([]string, error) {
```
Comment (hferentschik) [inline pkg/gits/git_cli.go]: _prefix_ seems to be a bit generic as a function parameter name in this context. Maybe `remoteBranchFilter`, `remoteRefPrefix` or something similar which comes out of the git domain

**Comment by mrageh** [comment]
i've found testing in JX to be a difficult, so I've skipped it for this PR. There's a ton of setup to do and at this moment in time I think I can use my time to better learn Golang and Kubernetes

**Pull request merged**